source/val/ValidationState.cpp

```diff
@@ -357,4 +365,91 @@ spv_result_t ValidationState_t::RegisterFunctionEnd() {
   return SPV_SUCCESS;
 }
 
+ValidationState_t::UseDefTracker::UseDefTracker(ValidationState_t* state)
+    : module_(state) {}
+
+void ValidationState_t::UseDefTracker::AddDef(const spv_id_info_t& def) {
+  all_defs_[def.id] = def;
+  if (module_->in_block()) {
```

Comment:
The spv_id_info_t already has the ID of the defining block, if any.  So you can just check that instead of poking at the module_ member.   That would be more self-contained.